source/engine/graphics/api/vulkan/Lighting.cpp

```diff
@@ -1,118 +1,61 @@
 #include "Lighting.h"
-#include "FrameManager.h"
-#include "GpuAllocator.h"
 #include "RenderGraph.h"
-#include "Swapchain.h"
-#include "core/GpuOptions.h"
-#include "ecs/Registry.h"
 #include "graphics/PointLight.h"
-#include "renderpasses/RenderPass.h"
-#include "shaders/ShaderDescriptorSets.h"
 #include "shaders/ShaderResources.h"
+#include "shaders/ShadowMapShaderResource.h"
 
-#include <array>
-
-namespace
-{
-inline static const std::string ShadowMappingPassId = "Shadow Mapping Pass";
-}
+#include <algorithm>
 
 namespace nc::graphics
 {
 Lighting::Lighting(Registry* registry,
-                   vk::Device device,
-                   GpuAllocator* allocator,
-                   Swapchain* swapchain,
                    RenderGraph* renderGraph,
-                   ShaderDescriptorSets* shaderDescriptorSets,
                    ShaderResources* shaderResources,
-                   Vector2 dimensions)
-    : m_registry{registry}, 
-      m_device{device}, 
-      m_allocator{allocator},
-      m_swapchain{swapchain},
-      m_renderGraph{renderGraph},
-      m_shaderDescriptorSets{shaderDescriptorSets},
+                   uint32_t maxPointLights)
+    : m_renderGraph{renderGraph},
       m_shaderResources{shaderResources},
-      m_dimensions{dimensions},
-      m_onCommitPointLightConnection{registry->OnCommit<PointLight>().Connect([this](graphics::PointLight&){OnCommitPointLight();})},
-      m_onRemovePointLightConnection{registry->OnRemove<PointLight>().Connect([this](Entity){OnRemovePointLight();})},
-      m_numShadowCasters{0u}
+      m_onCommitPointLightConnection{registry->OnCommit<PointLight>().Connect([this](graphics::PointLight&){OnCommitPointLightConnection();})},
+      m_onRemovePointLightConnection{registry->OnRemove<PointLight>().Connect([this](Entity){OnRemovePointLightConnection();})},
+      m_maxPointLights{maxPointLights}
 {
-}
+    std::vector<ShadowMapData> shadowMaps;
+    shadowMaps.reserve(m_maxPointLights);
```

Comment:
I imagine they're the same value, but you're guaranteed to get the allocation size you want by reserving based on `shadowMapPasses.size()`. As a bonus, I think you could then totally remove the `m_maxPointLights` member.